tests/common/include/tests/common/tools/particle_gun.hpp

```diff
@@ -57,31 +44,31 @@ struct particle_gun {
         std::vector<std::pair<dindex, intersection_type>> intersection_record;
 
         // Loop over all surfaces in the detector
+
+        const auto &mask_store = detector.mask_store();
+        const auto &tf_store = detector.transform_store();
+
         for (const auto &volume : detector.volumes()) {
             for (const auto &[sf_idx, sf] :
                  enumerate(detector.surfaces(), volume)) {
 
                 // Retrieve candidate from the surface
+                // NOTE: Change to interection_initialize
                 intersection_type sfi;
                 if constexpr (std::is_same_v<trajectory_t, detail::helix>) {
```

Comment:
it is needed because of helix intersection needs the tolerance of `1e-4`.

Reply:
But so should ultimately the ray, too

Reply:
The  ray was using the default tolerance value. I think you did that on purpose?

Reply:
If I remember correctly, I did that because the masks had different tolerance types and that was not easy to address. We can leave it like it is. The tolerances need to be addressed for the navigator in a distinct PR anyways